tests/Integration/Action/CancelActionTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\Payum\PayU\Tests\Integration\Action;
+
+use Answear\Payum\PayU\Action\CancelAction;
+use Answear\Payum\PayU\Request\OrderRequestService;
+use Answear\Payum\PayU\Tests\Payment;
+use Answear\Payum\PayU\Tests\Util\FileTestUtil;
+use Answear\Payum\PayU\ValueObject\Response\OrderCanceledResponse;
+use Answear\Payum\PayU\ValueObject\Response\OrderRetrieveResponse;
+use Payum\Core\Request\Cancel;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+class CancelActionTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function successTest(): void
+    {
+        $action = $this->getCancelAction(
+            OrderCanceledResponse::fromResponse(
+                FileTestUtil::decodeJsonFromFile(
+                    __DIR__ . '/../Request/data/orderCanceledResponse.json'
+                )
+            ),
+            OrderRetrieveResponse::fromResponse(
+                FileTestUtil::decodeJsonFromFile(
+                    __DIR__ . '/../Request/data/retrieveOrderResponse.json'
+                )
+            )
+        );
+
+        $payment = new Payment();
+        $payment->setDetails(FileTestUtil::decodeJsonFromFile(__DIR__ . '/data/detailsWithOrderId.json'));
+        $expectedBaseDetails = $this->getExpectedBaseDetails();
+        self::assertSame($expectedBaseDetails, $payment->getDetails());
```

Comment:
you just set the details. We need to check it again?

Reply:
it is not needed, removed